src/components/structures/MatrixChat.js

```diff
@@ -483,6 +486,12 @@ module.exports = React.createClass({
                 this._setPage(PageTypes.RoomDirectory);
                 this.notifyNewScreen('directory');
                 break;
+            case 'view_group':
+                const groupId = payload.group_id;
+                this.setState({currentGroupId: groupId});
+                this._setPage(PageTypes.GroupView);
```

Comment:
The above two lines could race but as they're both `setState`s I guess it's fine